instruqt-tracks/DEVELOPMENT.md

```diff
@@ -0,0 +1,46 @@
+# How to safely test Instruqt tracks
+
+Instruqt can be a bit finicky when it comes to testing tracks, because
+it appears internally to use track and challenge IDs and embeds them
+in your `track.yml` and `assigment.md` files when you push tracks. This
+can cause a problem when trying to make a test version of the track; if
+you are not careful you can confuse Instruqt or cause it to overwrite
+the production version of a track. 
+
+What follows is our standard flow and procedure for making a test 
+version of the track.
+
+# Standard Flow
+
+1. Create a Jira ticket for your work—all work should be logged and
+   documented in a Jira ticket anyways, and we use the ticket identifier
+   to provide context when committing changes. For this document, we're
+   going to assume you have a ticket number of `IL-99999`
+1. Ensure you are on the `master` branch of this repo (the current default
+   branch) and do not have uncommited changes. Pull the latest from 
+   `origin` (e.g. `git pull origin master`)
+1. Create a branch for this work, e.g. `git checkout -b IL-99999`
+1. In the track directory, run the command
+   `make jira=IL-99999 new_test_track`
+   - This removes the `track.yml` id and checksum fields, and all
+     of the `assignment.md` id fields
+   - It also changes the track slug so it starts with 
+     `wip-il-99999-<original slug`, and the track title to start
+     with `WIL IL-99999 - <Original Title>`
+   - The end result of this is that Instruqt will treat this like
+     an entirely independent track, as well as make it easy to
+     distinguish this as a test track
+1. Do whatever work you need to do. You can safely `instruqt track push`
+   to test your track
+1. When it becomes time to merge, change the track slug and title 
+   to remove the `wip-il-99999-` and `WIP IL-99999 - ` prefixes. If there
```

Comment:
If we put these mutations in a separate commit from other changes, we can make reverting them easier.

Reply:
Do you mean something like:

```
1. When it becomes time to merge, make a final commit with the track slug and title
   still changed but all other work done. Then change the track slug and title to remove
   the `wip-il-99999-` and `WIP IL-99999 - ` prefixes. If there are any track or challenge
   `id:` or `checksum:` fields, you should remove them; `make clean_id_and_checksums`
   will do that for you. This way if you need to continue work after the merge for any reason
   you can check out the branch and revert to that `N-1` commit.
```

I'd have to check and see what git/GitHub does if I push the branch to GitHub, then locally revert it to commit `N-1`, make a series of commits there, and then try to push it back up to the same branch. But in general, if this is what you mean, I think it's useful. Again, not blocking, but putting in the back of my mind for next time.